htslib/tbx.h

```diff
@@ -44,6 +44,7 @@ typedef struct tbx_conf_t {
     int32_t preset;
     int32_t sc, bc, ec; // seq col., beg col. and end col.
     int32_t meta_char, line_skip;
+    int64_t max_ref_len;
```

Comment:
Unfortunately this likely breaks the ABI, especially as `tbx_conf_t` is directly bunged into `tbx_t` below so expanding it will cause all the other `tbx_t` members to move.

The simplest solution here might be to remove the part that lets you set the maximum length, and just go with the hard-coding to 100G for now.  (The much more mind-boggling solution is to work out how to make `hts_idx_push()` expand `n_lvls` itself when the new item doesn't fit.  This is probably a [twenty pipe problem](https://en.wiktionary.org/wiki/three-pipe_problem).)

Reply:
Is this 100G the maximum length of a sequence or the total assembly ? The largest genome known so far is around 150 GB (_Paris japonica_).

Reply:
It's the largest length of an individual chromosome.

I thought about ABI, but concluded it's just extending a structure which isn't likely to be used in an array.  I didn't look very hard clearly as embedded it to the start of another struct is all of 2 lines lower!

It feels a bit poor to replace one arbitrary limit with another arbitrary limit, but I guess if it's large enough and we don't believe there are negative impacts of having it too large.  What does it actually do?  Maybe nothing if the levels don't get to be used due to the actual length not requiring them, but I'm hazy on this which worries me.

I can cull it back to a more minimal PR though if you don't wish to bump the ABI.

Reply:
I think I'd prefer not to break the ABI over this.